tools/version_docomentaion.py

```diff
@@ -0,0 +1,84 @@
+import subprocess
+import argparse
+import logging
+import shutil
+import os
+import re
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+logging.getLogger().setLevel(logging.INFO)
+
+REPO_URL = "https://github.com/openshift/assisted-service.git"
+valid_commit_regex = '^([A-Z]+-[0-9]+|#[0-9]+|merge|no-issue)'
+
+############################################################################################
+# This is used to generate the documentation of assisted-service between different versions.
+############################################################################################
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--from-version", help="From version to document", type=str)
+parser.add_argument("--to-version", help="To version to document", type=str)
+parser.add_argument("--documentation-dir", help="deployment yaml file to update", type=str,
+                    default=os.path.join(os.path.dirname(__file__), "../versions_documentation"))
+args = parser.parse_args()
+
+
+def main():
+    version_documentation_list = list()
+
+    if not os.path.exists(args.documentation_dir):
+        os.makedirs(args.documentation_dir)
+
+    os.mkdir("temp")
+    try:
+        git_logs_line = get_versions_log()
+        process_logs(git_logs_line, version_documentation_list)
+        write_documentation_to_file(version_documentation_list)
+    except Exception as ex:
+        logging.error('Failed to process versions documentation: {}'.format(ex))
+    finally:
+        shutil.rmtree("temp")
+
+
+def process_logs(git_logs_line, version_documentation_list):
+    for line in git_logs_line:
+        line = line.strip().decode("utf-8")
+        if is_line_metadata(line):
+            continue
+        line = line.strip('* ')
+
+        if re.match(valid_commit_regex, line):
```

Comment:
why are you checking if it's a valid commit regex? we already saw that we have problems when we cherry-pick commits in a branch - please verify that it's working on a branch

Reply:
that's exactly why I take only the validated lines, the metadata is not documented